### PR TITLE
Return-based error-handling for shell handlers

### PR DESCRIPTION
**Pull request by kushalsingh007**
-Updated version of branch used by @authmillenon ( continuation of his work )
-Removed whitespace errors
-Part of fix for #708

### CODE REVIEW

File: examples/rpl_udp/helper.c [hunk 1]
```diff
@@ -148,7 +150,16 @@ void rpl_udp_ignore(int argc, char **argv)
...
+    if (argc == 2) {
+        a = atoi(argv[1]);
+        printf("sending to transceiver (%u): %u\n", transceiver_pid, (*(uint8_t *)tcmd.data));
+        msg_send(&mesg, transceiver_pid, 1);
+
+        return 0;
+    }
+    else {
+        printf("Usage: %s <addr>\n", argv[0]);
+    }
```
Comment (gebart) [inline examples/rpl_udp/helper.c]: redundant check for number of arguments, already checked on L132

**Comment by authmillenon** [comment]
Needs rebase. It also reverts #2582 and I found the following missing ports:

```
$ git grep "void [^ ]*(int argc, char \*\*argv)"
cpu/native/startup.c:__attribute__((constructor)) static void startup(int argc, char **argv)
examples/rpl_udp/rpl_udp.h:void rpl_udp_init(int argc, char **argv);
examples/rpl_udp/rpl_udp.h:void rpl_udp_set_id(int argc, char **argv);
examples/rpl_udp/rpl_udp.h:void rpl_udp_dodag(int argc, char **argv);
examples/rpl_udp/rpl_udp.h:void udp_server(int argc, char **argv);
examples/rpl_udp/rpl_udp.h:void udp_send(int argc, char **argv);
examples/rpl_udp/rpl_udp.h:void rpl_udp_ignore(int argc, char **argv);
pkg/openwsn/0005-fixes-to-RIOT-adaption.patch:+void openwsn_start_thread(int argc, char **argv) {
pkg/openwsn/0005-fixes-to-RIOT-adaption.patch:+void openwsn_start_thread(int argc, char **argv);
sys/shell/commands/sc_lps331ap.c:void _get_lps331ap_init_handler(int argc, char **argv)
sys/shell/commands/sc_lps331ap.c:void _get_lps331ap_read_handler(int argc, char **argv)
sys/shell/commands/sc_netif.c:void _netif_list(int argc, char **argv)
sys/shell/commands/sc_netif.c:void _netif_send(int argc, char **argv)
sys/shell/commands/sc_netif.c:void _netif_addr(int argc, char **argv)
sys/shell/commands/sc_netif.c:void _netif_chan(int argc, char **argv)
sys/shell/commands/sc_netif.c:void _netif_pan(int argc, char **argv)
sys/shell/commands/sc_transceiver.c:void _transceiver_get_set_pan_handler(int argc, char **argv)
sys/shell/commands/shell_commands.c:extern void _rtc_handler(int argc, char **argv);
sys/shell/commands/shell_commands.c:void _netif_list(int argc, char **argv);
sys/shell/commands/shell_commands.c:void _netif_send(int argc, char **argv);
sys/shell/commands/shell_commands.c:void _netif_addr(int argc, char **argv);
sys/shell/commands/shell_commands.c:void _netif_chan(int argc, char **argv);
sys/shell/commands/shell_commands.c:void _netif_pan(int argc, char **argv);
```

**Comment by authmillenon** [comment]
(I made an error in the previous comment: I meant #2581)

**Comment by kushalsingh007** [comment]
@authmillenon  For your second last comment -
```
cpu/native/startup.c:__attribute__((constructor)) static void startup(int argc, char **argv)
pkg/openwsn/0005-fixes-to-RIOT-adaption.patch:+void openwsn_start_thread(int argc, char **argv) {
pkg/openwsn/0005-fixes-to-RIOT-adaption.patch:+void openwsn_start_th read(int argc, char **argv);
```

are the one's I was a little concerned about for the two patches I am not supposed to change them or do I have to resubmit the patch ?
for the cpu/native/startup.c code block , it used argc and **argv but then I on first though I think that they can be changed to int , as it uses ```usage_exit()```

Can you explain what exactly is meant by reverting #2581 , does that mean I resolved the merge conflicts while rebasing incorrectly previously ?

**Comment by authmillenon** [comment]
> are the one's I was a little concerned about for the two patches I am not supposed to change them or do I have to resubmit the patch ?

Just try it and compile `examples/default` with `USEPKG=openwsn`, if the patch target does not fail, we are good. As far as I remember git is quite capable of handling changed patch-files.

> for the cpu/native/startup.c code block , it used argc and **argv but then I on first though I think that they can be changed to int , as it uses usage_exit()

I think you must not concern yourself with this one. It's a normal UNIX process main (for the native process), so changing them might be harmful to the native port.

> Can you explain what exactly is meant by reverting #2581 , does that mean I resolved the merge conflicts while rebasing incorrectly previously ?

Obviously, your changes apply to the version of the commands before this. #2581 reduced the shell handlers for `MODULE_NG_NETIF` to `_netif_config` and `_netif_send`, with `_netif_config` working similar as `_net_if_ifconfig`, just with `ng_netif` instead of `net_if` ;-)

**Comment by kushalsingh007** [comment]
I tried compiling ```examples/default``` with ```USEPKG=openwsn``` however even without my changes to it , it failed to compile due to errors like ```‘TIMER_1’ undeclared ``` etc. And so I am skipping it in this my recent commit.

**Comment by authmillenon** [comment]
@thomaseichinger do you know what this is about?

**Comment by kushalsingh007** [comment]
```All is well``` and  @authmillenon  the work that you suggested has been  done(except for changing openwsn patch)  , can you suggest what part to do next.

File: examples/rpl_udp/helper.c [hunk 2]
```diff
@@ -148,7 +150,9 @@ void rpl_udp_ignore(int argc, char **argv)
     tcmd.transceivers = TRANSCEIVER_CC1100;
     tcmd.data = &a;
 
-    printf("sending to transceiver (%" PRIkernel_pid "): %u\n", transceiver_pid,
-           (*(uint8_t *)tcmd.data));
+    a = atoi(argv[1]);
+    printf("sending to transceiver (%u): %u\n", transceiver_pid, (*(uint8_t *)tcmd.data));
```
Comment (authmillenon) [inline examples/rpl_udp/helper.c]: Don't get the point of this change, nor how it is related to return based error-handling.

Comment (authmillenon) [inline examples/rpl_udp/helper.c]: (Moreover: the a is set in https://github.com/RIOT-OS/RIOT/pull/2602/files#diff-774299209354f6ffeb33ebb74a48b5c9R144 already).

File: examples/rpl_udp/udp.c [hunk 3]
```diff
@@ -93,9 +95,9 @@ static void *init_udp_server(void *arg)
 }
 
 /* UDP send command */
-void udp_send(int argc, char **argv)
+int udp_send(int argc, char **argv)
 {
-    int sock;
+    int sock, res;
```
Comment (authmillenon) [inline examples/rpl_udp/udp.c]: `res` is never set as far as I can see.

Comment (authmillenon) [inline examples/rpl_udp/udp.c]: Sorry, my mistake https://github.com/RIOT-OS/RIOT/pull/2602/files#diff-0de396c2e62be3e1c65c4f289e9f45f1R143 and https://github.com/RIOT-OS/RIOT/pull/2602/files#diff-0de396c2e62be3e1c65c4f289e9f45f1R149

File: sys/net/include/net_if.h [hunk 4]
```diff
@@ -442,7 +442,7 @@ int net_if_set_eui64(int if_id, net_if_eui64_t *eui64);
  * Sets the address of the transceiver associated with the given interface.
  *
  * @pre     Transceivers of this interface has to be initialized and
- *          transceiver thread has to be started.
+ *          transceiver thread ha/s to be started.
```
Comment (authmillenon) [inline sys/net/include/net_if.h]: ?

**Comment by authmillenon** [comment]
Apart from https://github.com/RIOT-OS/RIOT/pull/2602#discussion_r27079155 and https://github.com/RIOT-OS/RIOT/pull/2602#discussion_r27079476 I'm fine with this PR

**Comment by kushalsingh007** [comment]
@authmillenon : Made the changes you suggested in my latest commit.

**Comment by authmillenon** [comment]
For the future: please don't squash your changes immediately so it's easier for me to see your updates.

**Comment by authmillenon** [comment]
(moreover, travis is already kindof overheated, so I prefer if I can stop the build of out-dated commits ;-), which I can't if you overwrite them)

**Comment by authmillenon** [comment]
Apart from that: ACK, when Travis is happy. Please squash (yes, you can overwrite my old commit with your's ;-))

**Comment by kushalsingh007** [comment]
Done squashing . Thanks for letting me use your commit :)  ( Now let's see how long Travis takes :P )

**Comment by authmillenon** [comment]
Since this is a bigger change, I would prefer if we merge this before #2581 and #2582 ;).

**Comment by kushalsingh007** [comment]
Kindly restart the build for this one :)

**Comment by authmillenon** [comment]
I did already for a hundredth time ^^. There are either reachability issues with github or one of the apt-repositories. I'll keep the tab open :-)

**Comment by authmillenon** [comment]
And go.

**Comment by authmillenon** [comment]
@kushalsingh007 congratz for this one ;-)

**Comment by kushalsingh007** [comment]
:smiley: Thanks

**Pull request merged**